travis/travis_install.linux.sh

```diff
@@ -1,3 +1,31 @@
 # Linux installation
 
 virtualenv -p python${YCMD_PYTHON_VERSION} ${YCMD_VENV_DIR}
+
+if [[ ${VIMSCRIPT} ]]; then
+  if [[ -d /tmp/vim/.git ]]; then
+    pushd /tmp/vim
+    git fetch
+    if ! git diff --exit-code --quiet ..origin/master; then
```

Comment:
Are we deliberately checking out master? The question is whether we want to catch upstream changes in vim that break our code or if we want To catch changes in our code which break our code. I think the latter, so I think we should check out a particular commit or tag of vim and build that. We can update it regularly if we like, but we don't want flakiness caused by checking out vim head?